pkg/feature/manifest.go

```diff
@@ -59,57 +208,99 @@ func loadManifestsFrom(fsys fs.FS, path string) ([]manifest, error) {
 	return manifests, nil
 }
 
-func createManifestFrom(fsys fs.FS, path string) manifest {
+func CreateBaseManifestFrom(fsys fs.FS, path string) *baseManifest { //nolint:golint,revive //No need to export baseManifest.
 	basePath := filepath.Base(path)
-	m := manifest{
-		name:     basePath,
-		path:     path,
-		patch:    strings.Contains(basePath, ".patch"),
-		template: filepath.Ext(path) == ".tmpl",
-		fsys:     fsys,
+	m := &baseManifest{
+		name:  basePath,
+		path:  path,
+		patch: strings.Contains(basePath, ".patch"),
+		fsys:  fsys,
 	}
 
 	return m
 }
 
-func (m *manifest) targetPath() string {
-	return fmt.Sprintf("%s%s", m.path[:len(m.path)-len(filepath.Ext(m.path))], ".yaml")
+func CreateTemplateManifestFrom(fsys fs.FS, path string) *templateManifest { //nolint:golint,revive //No need to export templateManifest.
+	basePath := filepath.Base(path)
+	m := &templateManifest{
+		name:  basePath,
+		path:  path,
+		patch: strings.Contains(basePath, ".patch"),
+		fsys:  fsys,
+	}
+
+	return m
 }
 
-func (m *manifest) process(data interface{}) error {
-	manifestFile, err := m.open()
-	if err != nil {
-		return err
+func CreateKustomizeManifestFrom(path string, fsys filesys.FileSystem) *kustomizeManifest { //nolint:golint,revive //No need to export kustomizeManifest.
+	basePath := filepath.Base(path)
+	if basePath == KustomizationFile {
+		path = filepath.Dir(path)
 	}
-	defer manifestFile.Close()
-
-	content, err := io.ReadAll(manifestFile)
-	if err != nil {
-		return fmt.Errorf("failed to create file: %w", err)
+	if fsys == nil {
+		fsys = filesys.MakeFsOnDisk()
 	}
-
-	if !m.template {
-		// If, by convention, the file is not suffixed with `.tmpl` we do not need to trigger template processing.
-		// It's safe to return at this point.
-		m.processedContent = string(content)
-		return nil
+	m := &kustomizeManifest{
+		name: basePath,
+		path: path,
+		fsys: fsys,
 	}
 
-	tmpl, err := template.New(m.name).Funcs(template.FuncMap{"ReplaceChar": ReplaceChar}).Parse(string(content))
-	if err != nil {
-		return fmt.Errorf("failed to create file: %w", err)
-	}
+	return m
+}
 
-	var buffer bytes.Buffer
-	if err := tmpl.Execute(&buffer, data); err != nil {
-		return err
+// parsing helpers
+// isKustomizeManifest checks default filesystem for presence of kustomization file at this path.
+func isKustomizeManifest(path string) bool {
+	if filepath.Base(path) == KustomizationFile {
+		return true
 	}
+	_, err := os.Stat(filepath.Join(path, KustomizationFile))
+	return err == nil
+}
 
-	m.processedContent = buffer.String()
+func isTemplateManifest(path string) bool {
+	return strings.Contains(path, ".tmpl")
+}
 
-	return nil
+func convertToUnstructureds(resources string, objs []*unstructured.Unstructured) ([]*unstructured.Unstructured, error) {
```

Comment:
I wonder, do we really have to take `objs` as param? How about appending the returned slice of objs to the target slice directly?

Reply:
agreed, done here